dask_ml/model_selection/_hyperband.py

```diff
@@ -0,0 +1,323 @@
+                 dry_run=False, **kwargs):
+    start = time()
+    assert isinstance(X, np.ndarray)
+    assert y is None or isinstance(y, np.ndarray)
+    model, m = model_meta_future
+    if meta is None:
+        meta = m
+    while meta['iterations'] < meta['partial_fit_calls']:
+        if not dry_run:
+            model.partial_fit(X, y, **fit_params)
+        meta["iterations"] += 1
+    meta['mean_fit_time'] += time() - start
+    return model, meta
+
+
+def _score(model_and_meta, x, y, dry_run=False):
+    start = time()
+    model, meta = model_and_meta
+    #  seed = hash(((k, v) for k, v in meta.items())) % 2**32
+    #  rng = np.random.RandomState(seed)
+    if dry_run:
+        #  score = rng.rand()
+        score = 0
+    else:
+        # TODO: change this to dask_ml.get_scorer
+        score = model.score(x, y)
+    meta.update(score=score)
+    assert meta['iterations'] > 0
+    meta['mean_score_time'] += time() - start
+    meta['mean_test_score'] = score
+    return meta
+
+
+def _top_k(results, k=1, sort="score"):
+    res = sorted(results, key=lambda x: x[sort])[-k:]
+    assert len(res) == k
+    return res
+
+
+def _to_promote(result, completed_jobs, eta=None, asynchronous=None):
+    bracket_models = [r for r in completed_jobs
+                      if r['bracket_iter'] == result['bracket_iter'] and
+                      r['bracket'] == result['bracket']]
+
+    if not asynchronous:
+        bracket_completed = len(bracket_models) == result['num_models']
+        if bracket_completed:
+            to_keep = len(bracket_models) // eta
+            if to_keep <= 1:
+                return []
+            top_results = _top_k(bracket_models, k=to_keep, sort="score")
+            for job in top_results:
+                job["num_models"] = to_keep
+                job['partial_fit_calls'] *= eta
+                job['bracket_iter'] += 1
+            return top_results
+        return []
+    raise ValueError
+
+
+def _create_model(model, params, random_state=42):
+    model = clone(model).set_params(**params)
+    if 'random_state' in model.get_params():
+        model.set_params(random_state=random_state)
+    #  meta = {}  # right now no meta information
+    return model, None
+
+
+def _bracket_name(s):
+    return "bracket={s}".format(s=s)
+
+
+def _model_id(s, n_i):
+    return "bracket={s}-{n_i}".format(s=s, n_i=n_i)
+
+
+async def _hyperband(model, params, X, y, max_iter=None, eta=None,
+                     dry_run=False, fit_params={}, random_state=42,
+                     asynchronous=None):
+    client = default_client()
+    rng = check_random_state(random_state)
+    N, R, brackets = _get_nr(max_iter, eta=eta)
+    params = iter(ParameterSampler(params, n_iter=sum(N),
+                                   random_state=random_state))
+    if isinstance(model, Incremental):  # this should probably be handled elsewhere
+        model = model.estimator
```

Comment:
Remind me again why this needed to be unwrapped?

Reply:
We don't want the dask-ml Incremental estimator to be included in all of the tasks.  We want to send around the scikit-learn estimator instead.  This is analagous to how dask.dataframes create tasks that operate pandas dataframes.  Nesting gets complicated quickly.

Reply:
Do we want to be passing Incremental objects into Hyperband in the first place?  Or is it assumed that anything we pass in will be incremental?  Could we remove the use of `Incremental` generally from our examples for Hyperband?